### ISSUE TITLE
[turborepo] Reports successful build (with missing build files) when 'next build' hangs

### ISSUE THREAD

**Issue by jbgrunewald** [vercel/turbo#3954]
### What version of Turborepo are you using?

1.8.1

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

When running `turbo run build` the build will succeed but not correctly generate all of the build files. This occurs in our case when the `next build` command hangs indefinitely due to some other issue (something to do with sentry wrappers in our case).

### Expected Behavior

If a build would causes `next build` to hang and never completely generate the build files, I would expect this to result in a failure for turbo.

### To Reproduce

In our case the `next build` command will hang due to an incompatibility (yet to be determined) @sentry/nextjs and nextjs. I've included a minimal example of our next.config.js file which produces the hanging builds for us, and the corresponding dependency versions.

Deps:
```
    "@sentry/nextjs": "^7.38.0",
    "@sentry/tracing": "^7.38.0",
    "next": "13.1.6",
```

Example next.config.js:
```
const { withSentryConfig } = require("@sentry/nextjs");
const createNextRouteConfig = require("nextjs-routes/config");

const withRoutes = createNextRouteConfig();

const nextConfig = {
  reactStrictMode: true,
  swcMinify: false,
  images: {
    domains: [<redacted>],
  },
  transpilePackages: [<redacted>],
  },

  // Optional build-time configuration options
  sentry: {
    // See the 'Configure Source Maps' and 'Configure Legacy Browser Support'
    // sections below for information on the following options:
    //   - disableServerWebpackPlugin
    //   - disableClientWebpackPlugin
    //   - autoInstrumentServerFunctions
    //   - hideSourceMaps
    //   - widenClientFileUpload
    //   - transpileClientSDK
  }
};

const sentryWebpackPluginOptions = {
  // Additional config options for the Sentry Webpack plugin. Keep in mind that
  // the following options are set automatically, and overriding them is not
  // recommended:
  //   release, url, org, project, authToken, configFile, stripPrefix,
  //   urlPrefix, include, ignore

  silent: false, // Suppresses all logs
  // For all available options, see:
  // https://github.com/getsentry/sentry-webpack-plugin#options.
};

// Make sure adding Sentry options is the last code to run before exporting, to
// ensure that your source maps include changes from all other Webpack plugins
module.exports = withSentryConfig(
    withRoutes(nextConfig),
    sentryWebpackPluginOptions
);
```

The key difference between success and hang is wrapping with withSentryConfig. I'm sure there are other scenarios that would cause the build to hang, but this is the one I have at the moment. Sentry isn't really the core concern, rather the successful build report from turbo.

**Comment by nathanhammond** [comment]
Sentry's integration into `next.js` feels like a maze with exactly one working path through it. I wish you luck with that.

I've personally never seen Turborepo exit when the child process fails to exit. When this occurs is the `next build` still running according to your process list? Are you able to create a shareable reproduction?

I attempted reproduction but it runs to completion for me:

```
pnpm dlx create-turbo@alpha
pnpm add --save @sentry/nextjs
pnpm dlx @sentry/wizard -i nextjs
```

```
~/repos/triage/3954/apps/docs (main *)$ pnpm next build
info  - Linting and checking validity of types  

warn  - In order to be able to deminify errors, @sentry/nextjs creates sourcemaps and uploads them to the Sentry server. Depending on your deployment setup, this means your original code may be visible in browser devtools in production. To prevent this, set hideSourceMaps to true in the sentry options in your next.config.js. To disable this warning without changing sourcemap behavior, set hideSourceMaps to false. (In @sentry/nextjs version 8.0.0 and beyond, this option will default to true.) See https://webpack.js.org/configuration/devtool/ and https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#use-hidden-source-map for more information.


warn  - In order to be able to deminify errors, @sentry/nextjs creates sourcemaps and uploads them to the Sentry server. Depending on your deployment setup, this means your original code may be visible in browser devtools in production. To prevent this, set hideSourceMaps to true in the sentry options in your next.config.js. To disable this warning without changing sourcemap behavior, set hideSourceMaps to false. (In @sentry/nextjs version 8.0.0 and beyond, this option will default to true.) See https://webpack.js.org/configuration/devtool/ and https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#use-hidden-source-map for more information.

info  - Compiled successfully
info  - Collecting page data  
info  - Generating static pages (2/2)
info  - Finalizing page optimization  

Route (pages)                              Size     First Load JS
┌ ○ /                                      346 B           108 kB
├ λ /404                                   365 B           108 kB
└ ○ /sentry_sample_error                   1.56 kB         109 kB
+ First Load JS shared by all              107 kB
  ├ chunks/framework-685b3dd75a901335.js   45.3 kB
  ├ chunks/main-e36ed602e49ec1b3.js        27.1 kB
  ├ chunks/pages/_app-e130e90e744294fc.js  33.8 kB
  └ chunks/webpack-eb926854a35f361d.js     996 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)

info  - Creating an optimized production build .
```

**Comment by jbgrunewald** [comment]
@nathanhammond Thanks for trying to take a look. I'll see if I can find some time to go back to our broken version and try to gather some more information. It may be possible to get a minimum reproduction. I'd be curious to see the branch you tried with, which might be a good starting point.

Our issues were certainly due to the nuances of integrating Sentry, and in our case removing the "swcMinify: false" option from the nextConfig allowed the build to complete normally. What I observed was that there was a very clear error when running `next dev` in the app directory that didn't happen when we ran `turbo run dev` from the repo root.
